drivers/dac/dac_stm32.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <drivers/dac.h>
+#include <device.h>
+#include <kernel.h>
+#include <init.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_DAC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dac_stm32);
+
+#include <clock_control/stm32_clock_control.h>
+
+/* 2 channels per DAC, first one named 1 */
+#define STM32_CHANNEL_COUNT		2
+#define STM32_FIRST_CHANNEL		1
+
+#define CHAN(n)		LL_DAC_CHANNEL_##n
+static const u32_t table_channels[] = {
+	CHAN(1),
+	CHAN(2),
```

Comment:
Please define under `#ifdef DAC_CHANNEL2_SUPPORT` or dts exacted flag.

Reply:
Found that ll_dac.h defines LL_DAC_CHANNEL_2 only if it actually exists in hardware, so I'll use that as a flag.

In general, I'm not entirely sure how DTS and Kconfig should be configured to reflect actual features of a specific MCU. I see that peripherals appear in menuconfig even if the MCU doesn't support it (e.g. CAN in case of STM32L0). They are not selected by default, but what happens if a user enables it?
In our case, if someone enables DAC_3 even though it doesn't exist in hardware, we get a compile error because the defines are not found. But shouldn't we better prevent selecting an impossible configuration at all in menuconfig?

Reply:
> I see that peripherals appear in menuconfig even if the MCU doesn't support it (e.g. CAN in case of STM32L0). They are not selected by default, but what happens if a user enables it?

This is indeed an issue, ifnot supported by SoC, it should not be visible. It shouldbe fixed.

> But shouldn't we better prevent selecting an impossible configuration at all in menuconfig?

DTS will provide a more accurate way to declare this, but there are still few hickups to get it fully supported

Reply:
Ok, so this seems to be a more general issue then if it affects many other peripheral drivers. Can we keep it as it is (i.e. similar to other drivers) here and discuss visibility of peripherals in DTS in general in a different issue?

Reply:
Sure, this is fine as is.